### ISSUE TITLE
Class Imbalance and Model Initialization

### ISSUE THREAD

**Issue by pasandrei** [pasandrei/MIRPR-pedestrian-and-vehicle-detection-SSDLite#22]
Binary classification models are by default initialized to have equal probability of outputting either y = −1 or 1. Under such an initialization, in the presence of class imbalance, the loss due to the frequent class can dominate total loss and cause instability in early training. To counter this,
we introduce the concept of a ‘prior’ for the value of p estimated by the model for the rare class (foreground) at the start of training. We denote the prior by π and set it so that the model’s estimated p for examples of the rare class is low, e.g. 0.01. We note that this is a change in model initialization (see §4.1) and not of the loss function. We found this to improve training stability for both the cross entropy and focal loss in the case of heavy class imbalance.

arXiv:1708.02002 [cs.CV]
